RHDEVDOCS-5247: Documentation for Pipelines Resolvers (#60482)
openshift/openshift-docs: modules/op-resolver-git.adoc

// This module is included in the following assembly:
//
// // *openshift_pipelines/remote-pipelines-tasks-resolvers.adoc

:_content-type: PROCEDURE
[id="resolver-git-specify_{context}"]
= Specifying a remote pipeline or task using the Git resolver

When creating a pipeline run, you can specify a remote pipeline from a Git repository. When creating a pipeline or a task run, you can specify a remote task from a Git repository.

.Prerequisites

> * If you want to use teh authenticated SCM API, you configured the authenticated Git connection for the Git resolver.

Reviewer: ```suggestion
* If you want to use the authenticated SCM API, you configured the authenticated Git connection for the Git resolver.
```

(Also in case you don't know, don't use the "Commit suggestion" button to accept this change - make the change manually)

// This module is included in the following assembly:
//
// // *openshift_pipelines/remote-pipelines-tasks-resolvers.adoc

:_content-type: PROCEDURE
[id="resolver-git-specify_{context}"]
= Specifying a remote pipeline or task using the git resolver

When creating a pipeline run, you can specify a remote pipeline from a Git repository. When creating a pipeline or a task run, you can specify a remote task from a Git repository.

.Prerequisites

> * If you want to use the authenticated SCM API, you configured the authenticated Git connection for the git resolver.

Reviewer 2: ```suggestion
* If you want to use the authenticated SCM API, you must configure the authenticated Git connection for the Git resolver.
```

.Procedure

. To specify a remote pipeline or task from a Git repository, use the following reference format in the `pipelineRef` or `taskRef` spec:
+
[source,yaml]
----
# ...
  resolver: git
  params:
  - name: url
    value: <git-repo-url>
  - name: revision
    value: <branch-name>
  - name: pathInRepo

>     value: <path-in-repo>

Reviewer 2: same comment as before about formatting replaceable values with `_` not `-`

# ...
----
+
.Supported parameters for the git resolver
|===
| Parameter | Description | Example value

| `url`
| The URL of the repository, when using anonymous cloning.
| `+https://github.com/tektoncd/catalog.git+`

| `repo`
| The repository name, when using the authenticated SCM API.
| `test-infra`

| `org`
| The organization for the repository, when using the authenticated SCM API.
| `tektoncd`

| `revision`
| The Git revision in the repository. You can specify a branch name, a tag  name, or a commit SHA hash.
| `aeb957601cf41c012be462827053a21a420befca`

> `main`

Reviewer: These examples don't get put onto different lines like you intended

// This module is included in the following assembly:
//
// // *openshift_pipelines/remote-pipelines-tasks-resolvers.adoc

:_content-type: PROCEDURE
[id="resolver-git_{context}"]
= Specifying a remote pipeline or task from a Git repository

Using the Git resolver, you can specify a remote pipeline or task from a Git repostory. The repository must contain a YAML file that defines the pipeine or task.

The Git resolver can access a repository either by cloning it anonymously or else by using the authenticated SCM API.

For the authenticated API, you can use Git repoisitory providers that are supported by the link:https://github.com/jenkins-x/go-scm/tree/main/scm/driver[`go-scm` library].

> Not all `go-scm` implementations have been tested with the Git resolver, but the following providers are known to work:

Reviewer 3: We may want to highlight a shortcoming today which is, it can only be configured for the entire cluster, making it a bit "brittle" if you want to have different team and different private repository to pull from (because you would need to create a user that has access to all).

cc @chmouel

Author: I believe I did this one.

// This module is included in the following assembly:
//
// // *openshift_pipelines/remote-pipelines-tasks-resolvers.adoc

:_content-type: PROCEDURE
[id="resolver-git_{context}"]
= Specifying a remote pipeline or task from a Git repository

> Using the Git resolver, you can specify a remote pipeline or task from a Git repostory. The repository must contain a YAML file that defines the pipeine or task.

Reviewer 4: typo - pipeine

The Git resolver can access a repository either by cloning it anonymously or else by using the authenticated SCM API.

For the authenticated API, you can use Git repository providers that are supported by the link:https://github.com/jenkins-x/go-scm/tree/main/scm/driver[`go-scm` library].
Not all `go-scm` implementations have been tested with the Git resolver, but the following providers are known to work:

* github.com and GitHub Enterprise
* gitlab.com and self-hosted Gitlab
* Gitea
* BitBucket Server
* BitBucket Cloud

[NOTE]
====

> You can configure only one Git connection using the authenticated SCM API for your cluster. This connection becomes available to all users of the cluster.

Reviewer 5: i would have add that you then share the token access to all users and may represent a security fix!

but i let the decision to someone else if we really need to make it scary like that cc @vdemeester

To edit the `TektonConfig` custom resource, enter the following command:

[source,bash]
----
oc edit TektonConfig config
----

In the `TektonConfig` custom resource, in the `pipeline.git-resolver-config` spec, specify  configuration for the authenticated SCM API, as in the following example:

[source,yaml]
----
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  pipeline:
    git-resolver-config:
      scm-type: github
      server-url: api.internal-github.com
      api-token-secret-name: github-auth-secret
      api-token-secret-key: github-auth-key
      api-token-secret-namespace: github-auth-namespace
----

.Parameters for configuring the Git resolver
|===
| Parameter | Description | Example value

| `default-revision`
| The default git revision to use if none is specified
| `main`

| `fetch-timeout`
| The maximum time any single git clone resolution may take. {pipelines-title} also enforces a global maximum timeout of 1 minute on all resolution requests.
| `1m`, `2s`, `700ms`

| `default-url`

> | The default git repository URL for anonymous cloning if none is specified.

Reviewer 6: ```suggestion
| The default Git repository URL for anonymous cloning if none is specified.
```

| `+https://github.com/tektoncd/catalog.git+`

| `scm-type`
| The SCM provider type. Required for using the authenticated SCM API.
| `github`, `gitlab`, `gitea`, `bitbucketcloud`, `bitbucketserver`

| `server-url`

> | The base URL for use with the authenticated SCM API. Not required if using `\github.com`, `\gitlab.com`, or BitBucket Cloud

Reviewer 6: ```suggestion
| The base URL for use with the authenticated SCM API. Not required if using `\github.com`, `\gitlab.com`, or BitBucket Cloud.
```

| `api.internal-github.com`

| `api-token-secret-name`

> | The name of the {OCP} secret that contains the SCM provider API token. Required for using the authenticated SCM API.

Reviewer 4: OCP not rendered correctly
